vllm/v1/worker/gpu_model_runner.py

```diff
@@ -2332,6 +2311,20 @@ def _bookkeeping_sync(
             req_state = self.requests[req_id]
             req_state.output_token_ids.extend(sampled_ids)
 
+            cu_num_accepted_tokens.append(cu_num_accepted_tokens[-1] + len(sampled_ids))
+
+        cu_num_generated_tokens = (
+            cu_num_accepted_tokens if spec_decode_metadata else None
```

Comment:
Could we put this check inside the loop to avoid building the list in the None case? And also no need to do it if `logprobs_tensors is None` right?